sqlmesh/core/engine_adapter/base.py

```diff
@@ -944,7 +946,11 @@ def scd_type_2(
                     "deleted",
                     exp.select(*[f"static.{col}" for col in columns_to_types])
                     .from_("static")
-                    .join("latest", using=unique_key, join_type="left")
+                    .join(
+                        "latest",
+                        on=" AND ".join(f"static.{col} = latest.{col}" for col in unique_key),
```

Comment:
@tobymao @GeorgeSittas Should SQLGlot be updated to correctly transpile using for MSSQL?

Reply:
Not sure what part of the transpilation is incorrect; what is the current vs expected behavior in SQLGlot?

Reply:
@treysp Can you explain why you replaced all the `using` with `on` throughout this PR?

Reply:
MSSQL doesn't support `USING`. `USING` can't be transpiled into `ON` since the semantics of the two is different.

Reply:
>MSSQL doesn't support USING. USING can't be transpiled into ON since the semantics of the two is different.

Is that right? Isn't USING just ON but with both columns qualified to match the two tables being joined?